in_toto/runlib.go

```diff
@@ -66,7 +72,10 @@ the following format:
 If recording an artifact fails the first return value is nil and the second
 return value is the error.
 */
-func RecordArtifacts(paths []string) (map[string]interface{}, error) {
+func RecordArtifacts(paths []string, depth uint) (map[string]interface{}, error) {
+	if depth > 10 {
```

Comment:
Why 10?

Reply:
10 is just an artificial number I have inherit from the former PR. We should actually think about a better number here or at least a good description, why we choose 10 and not 9 or 11 or 9000.

Reply:
We could set the go runtime stack size here. Problem: It's different for the architecture. On Linux 64 bit it's usually 1GB.

But I have maybe a better solution, but this would need some more work for this PR.
What about using a history of visited symlinks? We could just add a visited symlink to a slice or hashset (`map[string]struct{}` for lookup time of O(1)). If we encounter a symlink twice, we just ignore it. What do you think about this, @lukpueh?

Reply:
I had to try this. We are now using a symlink history via a global hashset variable. This has better performance, because we don't need to cycle n times, before detecting a symlink.
Instead of we are checking if we have seen this symlink already, if so, we will break the recursion and error our Symlink Cycle error.

Feel free to check it out.

Reply:
I was hoping you would implement something like this, because I think the depth parameter would have failed detecting some indirect circles. Thanks for putting more efforts into this, @shibumi! 🥇